packages/react/src/themes/teams/components/Label/labelVariables.ts

```diff
@@ -1,9 +1,25 @@
 import { pxToRem } from '../../../../lib'
-import { SiteVariablesPrepared, ColorSchemeMapping } from '../../../types'
-import { extendColorScheme } from '../../../colorUtils'
+import {
+  SiteVariablesPrepared,
+  StrictColorSchemeMapping,
+  StrictColorScheme,
+  ComponentAreaName,
+} from '../../../types'
+import { extendColorScheme, pickValuesFromColorScheme } from '../../../colorUtils'
+import { TeamsColorNames } from '../../colors'
+
+// TODO extract this to a util
+const tuple = <T extends ComponentAreaName>(...args: T[]) => args
+const labelColorComponentAreas = tuple('background', 'foreground')
+type LabelColorComponentAreas = typeof labelColorComponentAreas[number]
+
+type LabelColorSchemeMapping = StrictColorSchemeMapping<
+  StrictColorScheme<LabelColorComponentAreas>,
+  TeamsColorNames
+>
 
 export interface LabelVariables {
-  colorScheme: ColorSchemeMapping
+  colorScheme: LabelColorSchemeMapping
```

Comment:
This may be a problem when we are using here some base theme interface and want to extend this, as this prop should have a different type..

Reply:
just to be sure this is not overlooked - could you, please, suggest an example of this (like, some pseudo-code)? Because at the moment I think that it would be normal to extend this Teams interface in that case - and semantically that would mean that now this colors' set is dependent on colors object (or, as equivalent, its type) provided by base theme.

Reply:
Sure, let me provide an example. The text component in the base themes, specifies some color names, that may won't exists in the Teams theme (like blue for example). That means, that the interface that we use here will be different. For example:

```
//base theme:
interface TextVariables {
  colorScheme: ColorSchemeMapping<ColorScheme, ColorNames>
}

//Teams theme:
interface TextVariables {
  colorScheme: ColorSchemeMapping<ColorScheme, TeamsColorNames>
}
```